File: problems_list/剑指OfferII049.从根节点到叶节点的路径数字之和/剑指OfferII049-从根节点到叶节点的路径数字之和.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var dfs=function(root, preSum){
    if(root==null){
        return 0;
    }
    var sum=preSum*10+root.val;
    if(root.left==null && root.right==null){
        return sum;
    }else{
        return dfs(root.left, sum)+dfs(root.right, sum);
    }
}
var sumNumbers = function(root) {
    return dfs(root, 0);
};

File: problems_list/剑指OfferII049.从根节点到叶节点的路径数字之和/剑指OfferII049-从根节点到叶节点的路径数字之和.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int dfs(TreeNode* root, int preSum){
        if(!root){
            return 0;
        }
        int sum=root->val+preSum*10;
        if(!root->left && !root->right){
            return sum;
        }else{
            return dfs(root->left, sum)+dfs(root->right, sum);
        }
    }
    int sumNumbers(TreeNode* root) {
        return dfs(root, 0);
    }
};